src/panel/direction/road_map_panel.js

```diff
@@ -1,136 +1,35 @@
-import Panel from '../../libs/panel';
-import roadMapTemplate from '../../views/direction/road_map.dot';
-import Device from '../../libs/device';
-import RoadMapPreviewPanel from './road_map_preview';
-import Telemetry from '../../libs/telemetry';
-import { openShareModal } from 'src/modals/ShareModal';
-import { formatDuration, formatDistance } from 'src/libs/route_utils';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import RouteResult from './RouteResult';
 
 export default class RoadMapPanel {
-  constructor(onOpen, onClose) {
-    this.onOpen = onOpen;
-    this.onClose = onClose;
-    this.previewRoadMap = new RoadMapPreviewPanel(this.distance);
-    this.showRoute = true;
-    this.panel = new Panel(this, roadMapTemplate);
-    this.routes = [];
-    this.isMobile = Device.isMobile;
-    this.placeholder = false;
-    this.error = false;
-    this.origin = null;
-
-    listen('select_road_map', i => {
-      this.toggleRoute(i);
-    });
+  constructor(openMobilePreview) {
+    this.onOpenMobilePreview = openMobilePreview;
   }
 
+  getContainer = () => document.getElementById('react_itinerary_result');
+
   setRoad(routes, vehicle, origin) {
-    if (origin) {
-      this.origin = origin;
-    }
-    this.routes = routes.map(roadStep => {
-      return roadStep;
-    });
-    this.vehicle = vehicle;
-    this.error = false;
-    this.panel.update();
+    ReactDOM.render(
+      <RouteResult
+        routes={routes}
+        vehicle={vehicle}
+        origin={origin.getInputValue()}
```

Comment:
`origin` may be null here. 
For example when called on blur with empty input field: https://github.com/QwantResearch/erdapfel/pull/364/files#diff-f91eb1ec5c604bd61eb88e3aa50ed7f6R79

Reply:
Fixed it, thanks.